From ce0e4cf60033eb7e15266b329183f945f2a82c61 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Mon, 2 Dec 2019 10:28:44 +0000
Subject: [PATCH] spidev: Completely disable the DT warning

The upstream SPI and DT maintainers are completely opposed to declaring
in Device Tree that an SPI CS line is to be managed by the spidev
driver, even though the facility is useful on a hobbyist device like a
Raspberry Pi where arbitrary devices can be attached, and the
alternative to DT declaration (spi_board_info) has been almost entirely
rendered obsolete by DT.

Continue to override their objections by disabling the warning.

See: https://github.com/raspberrypi/linux/issues/3361
     https://github.com/raspberrypi/linux/issues/1054

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 drivers/spi/spidev.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -728,6 +728,7 @@ static int spidev_probe(struct spi_devic
 	 * compatible string, it is a Linux implementation thing
 	 * rather than a description of the hardware.
 	 */
+	if (0) /* Disable the warning - this feature is too useful */
 	WARN(spi->dev.of_node &&
 	     of_device_is_compatible(spi->dev.of_node, "spidev"),
 	     "%pOF: buggy DT: spidev listed directly in DT\n", spi->dev.of_node);
